--- a/frameworks/av/media/libcedarx/external/Android.bp
+++ b/frameworks/av/media/libcedarx/external/Android.bp
@@ -1,128 +1,160 @@
 cc_prebuilt_library_shared {
     name: "libadecoder",
-    srcs: ["lib32/arm-linux-androideabi/libadecoder.so"],
-
-    compile_multilib: "32",
+    arch: {
+        arm: { srcs: ["lib32/arm-linux-androideabi/libadecoder.so"] },
+        arm64: { srcs: ["lib64/aarch64-linux-gnu/libadecoder.so"] },
+    },
+    compile_multilib: "both",
     shared_libs: ["libc", "libdl", "liblog", "libm"],
 }
 
 cc_prebuilt_library_shared {
     name: "libaencoder",
-    srcs: ["lib32/arm-linux-androideabi/libaencoder.so"],
-
-    compile_multilib: "32",
+    arch: {
+        arm: { srcs: ["lib32/arm-linux-androideabi/libaencoder.so"] },
+        arm64: { srcs: ["lib64/aarch64-linux-gnu/libaencoder.so"] },
+    },
+    compile_multilib: "both",
     shared_libs: ["libc", "libdl", "liblog", "libm"],
 }
 
 cc_prebuilt_library_shared {
     name: "libaw_aacdec",
-    srcs: ["lib32/arm-linux-androideabi/libaw_aacdec.so"],
-
-    compile_multilib: "32",
+    arch: {
+        arm: { srcs: ["lib32/arm-linux-androideabi/libaw_aacdec.so"] },
+        arm64: { srcs: ["lib64/aarch64-linux-gnu/libaw_aacdec.so"] },
+    },
+    compile_multilib: "both",
     shared_libs: ["libc", "libdl", "liblog", "libm"],
 }
 
 cc_prebuilt_library_shared {
     name: "libaw_alacdec",
-    srcs: ["lib32/arm-linux-androideabi/libaw_alacdec.so"],
-
-    compile_multilib: "32",
+    arch: {
+        arm: { srcs: ["lib32/arm-linux-androideabi/libaw_alacdec.so"] },
+        arm64: { srcs: ["lib64/aarch64-linux-gnu/libaw_alacdec.so"] },
+    },
+    compile_multilib: "both",
     shared_libs: ["libc", "libdl", "liblog", "libm"],
 }
 
 cc_prebuilt_library_shared {
     name: "libaw_amrdec",
-    srcs: ["lib32/arm-linux-androideabi/libaw_amrdec.so"],
-
-    compile_multilib: "32",
+    arch: {
+        arm: { srcs: ["lib32/arm-linux-androideabi/libaw_amrdec.so"] },
+        arm64: { srcs: ["lib64/aarch64-linux-gnu/libaw_amrdec.so"] },
+    },
+    compile_multilib: "both",
     shared_libs: ["libc", "libdl", "liblog", "libm"],
 }
 
 cc_prebuilt_library_shared {
     name: "libaw_mp3dec",
-    srcs: ["lib32/arm-linux-androideabi/libaw_mp3dec.so"],
-
-    compile_multilib: "32",
+    arch: {
+        arm: { srcs: ["lib32/arm-linux-androideabi/libaw_mp3dec.so"] },
+        arm64: { srcs: ["lib64/aarch64-linux-gnu/libaw_mp3dec.so"] },
+    },
+    compile_multilib: "both",
     shared_libs: ["libc", "libdl", "liblog", "libm"],
 }
 
 cc_prebuilt_library_shared {
     name: "libaw_apedec",
-    srcs: ["lib32/arm-linux-androideabi/libaw_apedec.so"],
-
-    compile_multilib: "32",
+    arch: {
+        arm: { srcs: ["lib32/arm-linux-androideabi/libaw_apedec.so"] },
+        arm64: { srcs: ["lib64/aarch64-linux-gnu/libaw_apedec.so"] },
+    },
+    compile_multilib: "both",
     shared_libs: ["libc", "libdl", "liblog", "libm"],
 }
 
 cc_prebuilt_library_shared {
     name: "libaw_dsddec",
-    srcs: ["lib32/arm-linux-androideabi/libaw_dsddec.so"],
-
-    compile_multilib: "32",
+    arch: {
+        arm: { srcs: ["lib32/arm-linux-androideabi/libaw_dsddec.so"] },
+        arm64: { srcs: ["lib64/aarch64-linux-gnu/libaw_dsddec.so"] },
+    },
+    compile_multilib: "both",
     shared_libs: ["libc", "libdl", "liblog", "libm"],
 }
 
 cc_prebuilt_library_shared {
     name: "libaw_flacdec",
-    srcs: ["lib32/arm-linux-androideabi/libaw_flacdec.so"],
-
-    compile_multilib: "32",
+    arch: {
+        arm: { srcs: ["lib32/arm-linux-androideabi/libaw_flacdec.so"] },
+        arm64: { srcs: ["lib64/aarch64-linux-gnu/libaw_flacdec.so"] },
+    },
+    compile_multilib: "both",
     shared_libs: ["libc", "libdl", "liblog", "libm"],
 }
 
 cc_prebuilt_library_shared {
     name: "libaw_g729dec",
-    srcs: ["lib32/arm-linux-androideabi/libaw_g729dec.so"],
-
-    compile_multilib: "32",
+    arch: {
+        arm: { srcs: ["lib32/arm-linux-androideabi/libaw_g729dec.so"] },
+        arm64: { srcs: ["lib64/aarch64-linux-gnu/libaw_g729dec.so"] },
+    },
+    compile_multilib: "both",
     shared_libs: ["libc", "libdl", "liblog", "libm"],
 }
 
 cc_prebuilt_library_shared {
     name: "libaw_oggdec",
-    srcs: ["lib32/arm-linux-androideabi/libaw_oggdec.so"],
-
-    compile_multilib: "32",
+    arch: {
+        arm: { srcs: ["lib32/arm-linux-androideabi/libaw_oggdec.so"] },
+        arm64: { srcs: ["lib64/aarch64-linux-gnu/libaw_oggdec.so"] },
+    },
+    compile_multilib: "both",
     shared_libs: ["libc", "libdl", "liblog", "libm"],
 }
 
 cc_prebuilt_library_shared {
     name: "libaw_wavdec",
-    srcs: ["lib32/arm-linux-androideabi/libaw_wavdec.so"],
-
-    compile_multilib: "32",
+    arch: {
+        arm: { srcs: ["lib32/arm-linux-androideabi/libaw_wavdec.so"] },
+        arm64: { srcs: ["lib64/aarch64-linux-gnu/libaw_wavdec.so"] },
+    },
+    compile_multilib: "both",
     shared_libs: ["libc", "libdl", "liblog", "libm"],
 }
 
 cc_prebuilt_library_shared {
     name: "libaw_atrcdec",
-    srcs: ["lib32/arm-linux-androideabi/libaw_atrcdec.so"],
-
-    compile_multilib: "32",
+    arch: {
+        arm: { srcs: ["lib32/arm-linux-androideabi/libaw_atrcdec.so"] },
+        arm64: { srcs: ["lib64/aarch64-linux-gnu/libaw_atrcdec.so"] },
+    },
+    compile_multilib: "both",
     shared_libs: ["libc", "libdl", "liblog", "libm"],
 }
 
 cc_prebuilt_library_shared {
     name: "libaw_cookdec",
-    srcs: ["lib32/arm-linux-androideabi/libaw_cookdec.so"],
-
-    compile_multilib: "32",
+    arch: {
+        arm: { srcs: ["lib32/arm-linux-androideabi/libaw_cookdec.so"] },
+        arm64: { srcs: ["lib64/aarch64-linux-gnu/libaw_cookdec.so"] },
+    },
+    compile_multilib: "both",
     shared_libs: ["libc", "libdl", "liblog", "libm"],
 }
 
 cc_prebuilt_library_shared {
     name: "libaw_radec",
-    srcs: ["lib32/arm-linux-androideabi/libaw_radec.so"],
-
-    compile_multilib: "32",
+    arch: {
+        arm: { srcs: ["lib32/arm-linux-androideabi/libaw_radec.so"] },
+        arm64: { srcs: ["lib64/aarch64-linux-gnu/libaw_radec.so"] },
+    },
+    compile_multilib: "both",
     shared_libs: ["libc", "libdl", "liblog", "libm"],
 }
 
 cc_prebuilt_library_shared {
     name: "libaw_opusdec",
-    srcs: ["lib32/arm-linux-androideabi/libaw_opusdec.so"],
-
-    compile_multilib: "32",
+    arch: {
+        arm: { srcs: ["lib32/arm-linux-androideabi/libaw_opusdec.so"] },
+        arm64: { srcs: ["lib64/aarch64-linux-gnu/libaw_opusdec.so"] },
+    },
+    compile_multilib: "both",
     shared_libs: ["libc", "libdl", "liblog", "libm"],
 }
 
@@ -144,26 +176,32 @@
 
 cc_prebuilt_library_shared {
     name: "liblive555",
-    srcs: ["lib32/anougat70/liblive555.so"],
-
-    compile_multilib: "32",
+    arch: {
+        arm: { srcs: ["lib32/anougat70/liblive555.so"] },
+        arm64: { srcs: ["lib64/alp50/liblive555.so"] },
+    },
+    compile_multilib: "both",
     shared_libs: ["libc++", "libc", "libcutils", "libdl", "libm", "libstdc++"],
 }
 
 cc_prebuilt_library_shared {
     name: "librtp",
-    srcs: ["lib32/anougat70/librtp.so"],
-
-    compile_multilib: "32",
+    arch: {
+        arm: { srcs: ["lib32/anougat70/librtp.so"] },
+        arm64: { srcs: ["lib64/alp50/librtp.so"] },
+    },
+    compile_multilib: "both",
     shared_libs: ["libc++", "libc", "libcdx_base", "libcdx_stream", "libcutils",
     "libdl", "libm", "liblog"],
 }
 
 cc_prebuilt_library_shared {
     name: "libsubdecoder",
-    srcs: ["lib32/anougat70/libsubdecoder.so"],
-
-    compile_multilib: "32",
+    arch: {
+        arm: { srcs: ["lib32/anougat70/libsubdecoder.so"] },
+        arm64: { srcs: ["lib64/aarch64-linux-gnu/libsubdecoder.so"] },
+    },
+    compile_multilib: "both",
     shared_libs: ["libc++", "libc", "libcutils", "libdl", "liblog", "libm",
     "libutils"],
 }
