diff --git a/gpu_config.sh b/gpu_config.sh
index 268b291..4b613fd 100755
--- a/gpu_config.sh
+++ b/gpu_config.sh
@@ -88,6 +88,19 @@ gpu_setup_guest() {
     setprop ro.hardware.vulkan pastel
 }
 
+gpu_setup_opi03() {
+    echo "use Orange Pi Zero 3 H618 Mali-G31 mode"
+
+    chmod 666 /dev/mali0
+    chmod 666 /dev/cedar_dev
+    chmod 666 /dev/ion
+    chmod 666 /dev/g2d
+    setprop ro.hardware.egl mali
+    setprop ro.hardware.gralloc apollo
+    setprop ro.hardware.vulkan apollo
+    setprop debug.stagefright.ccodec 4
+}
+
 gpu_setup() {
     ## mode=(auto, host, guest)
     ## node=(/dev/dri/renderDxxx)
@@ -99,6 +112,8 @@ gpu_setup() {
         gpu_setup_host
     elif [ "$mode" = "guest" ]; then
         gpu_setup_guest
+    elif [ "$mode" = "opi03" ]; then
+        gpu_setup_opi03
     elif [ "$mode" = "auto" ]; then
          echo "use GPU auto mode"
          if setup_render_node; then
